api/library/python/iterm2/iterm2/alert.py

```diff
@@ -146,3 +147,165 @@ async def async_run(
+        """Returns the text field list [placeholder, default] """
+        return self.__text_field
+
+    @property
+    def checkboxes(self) -> typing.List[str]:
+        """Returns a list of checkbox strings """
+        return self.__checkboxes
+
+    @property
+    def checkbox_defaults(self) -> typing.List[int]:
+        """Returns a list of whether checkboxes are checked by default """
+        return self.__checkbox_defaults
+
+    @property
+    def combobox_items(self) -> typing.List[str]:
+        """Returns the list of strings that will populate the combobox"""
+        return self.__combobox_items
+
+    @property
+    def combobox_default(self) -> str:
+        """Returns the combox item that is initially chosen"""
+        return self.__combobox_default
+
+    @property
+    def window_id(self) -> typing.Optional[str]:
+        """Returns the window ID"""
+        return self.__window_id
+
+    def add_button(self, title: str):
+        """Adds a button to the end of the list of buttons."""
+        self.__buttons.append(title)
+
+    def add_checkboxes(self, items: typing.List[str],
+                       defaults: typing.List[int]):
+        """Adds an array of checkbox items to the end of the list of
+        checkboxes."""
+        idx = 0
+        for item_text in items:
+            self.__checkbox_defaults.append(defaults[idx])
+            self.__checkboxes.append(item_text)
+            idx = idx + 1
+
+    def add_checkbox_item(self, item_text: str, item_default: int = 0):
+        """Adds a checkbox to the end of the list of checkboxes."""
+        self.__checkbox_defaults.append(item_default)
+        self.__checkboxes.append(item_text)
+
+    def add_combobox(self, items: typing.List[str], default: str = ""):
+        """Adds a combobox item to the end of the list of __combobox_items."""
+        self.__combobox_default = default
+        for item_text in items:
+            self.__combobox_items.append(item_text)
+
+    def add_combobox_item(self, item_text: str, is_default: int = 0):
+        """Adds a combobox item to the end of the list of __combobox_items."""
+        self.__combobox_items.append(item_text)
+        if is_default == 1:
+            self.__combobox_default = item_text
+
+    def add_text_field(self, placeholder: str, default: str):
+        """add a text field to the alert """
+        self.__text_field.append(placeholder)
+        self.__text_field.append(default)
+
+    def set_width(self, width: int):
+        """ set the width of the poly modal alert (default is 300) """
+        self.__width = width
+
+    async def async_run(self,
+                        connection: iterm2.connection.Connection) -> dict[str, typing.Any]:
+        """Shows the poly modal alert.
+
+        :param connection: The connection to use.
+        :returns: an object containing the following keys -
+        - "button"  which will be a string with the name of the clicked button
```

Comment:
Define a new type for the return value rather than using a dict. This lets us have better type annotations and avoids the use of magic strings.